### PR TITLE
refactor: Remove v1 code

### LINKED ISSUE #631: Remove the obsolete V1 code
Remove the v1 code and refactor the v2 package.

### PR DESCRIPTION
**Pull request by weichou1229**
Close #631

## PR Checklist
Please check if your PR fulfills the following requirements:

- [x] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been added / updated (for bug fixes / features)

**If your build fails** due to your commit message not passing the build checks, please review the guidelines here: https://github.com/edgexfoundry/go-mod-core-contracts/blob/master/.github/Contributing.md.

## What is the new behavior?
Remove the v1 code and refactor the v2 package

## Does this PR introduce a breaking change?

- [ ] Yes
- [x] No

## New Imports

- [ ] Yes
- [x] No

## Specific Instructions
Are there any specific instructions or things that should be known prior to reviewing?

## Other information

### CODE REVIEW

**Comment by cloudxxx8** [comment]
@lenny-intel we need you suggestion of the package naming to put the constants.

File: clients/constants.go [hunk 1]
```diff
@@ -24,36 +24,6 @@ const (
 	CorrelationHeader    = "X-Correlation-ID" // Sets the key of the Correlation ID HTTP header
 )
 
-// Constants related to defined routes in the service APIs
```
Comment (lenny-intel) [inline clients/constants.go:27]: Does it make sense to move the remaining constants to the common constants? i.e. these remaining are not specific to using clients.

**Review by lenny-intel — CHANGES_REQUESTED** [review]

File: clients/constants.go (hunk 1 above)
Comment (lenny-intel) [reply clients/constants.go:27]: If agree to move these, please use separate commit so reviewing those changes can be done with out looking thru all the previous changes... ;-)

**Comment by lenny-intel** [comment]
> we need you suggestion of the package naming to put the constants.
> we name it `common` for now.

@cloudxxx8 , @weichou1229 , `common` package name works for me.

File: clients/constants.go (hunk 1 above)
Comment (weichou1229) [reply clients/constants.go:27]: Agree! Moved the remaining constants to the common package.

**Review by lenny-intel — APPROVED** [review]
LGTM

**Pull request merged**